### PR TITLE
feat(secureHeader): add Permissions-Policy header to secure headers middleware

### PR DESCRIPTION
**Pull request by kbkn3**
…iddleware

# Summary

Permissions-Policy was added to Secure Headers Middleware.
For more information on Permissions-Policy, please refer to the following.

EN: [Permissions-Policy | MDN Web Docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Permissions-Policy)
EN: [Permissions Policy - HTTP | MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Permissions_Policy)

JA: [Permissions-Policy - HTTP | MDN](https://developer.mozilla.org/ja/docs/Web/HTTP/Headers/Permissions-Policy)
JA: [Permissions Policy - HTTP | MDN](https://developer.mozilla.org/ja/docs/Web/HTTP/Permissions_Policy)

Documetn Policy, which is listed as a similar Header, is not a Security requirement and does not need to be addressed.

### The author should do the following, if applicable

- [x] Add tests
- [x] Run tests
- [x] `bun run format:fix && bun run lint:fix` to format the code
- [x] Add [TSDoc](https://tsdoc.org/)/[JSDoc](https://jsdoc.app/about-getting-started) to document the code

### CODE REVIEW

**Comment by yusukebe** [comment]
Hi @kbkn3 Thank you for the PR!

@watany-dev Can you review this?

File: src/middleware/secure-headers/secure-headers.ts [hunk 1]
```diff
@@ -255,6 +272,36 @@ function getCSPDirectives(
...
+        const allowlist = value.map((v) => (v === 'self' || v === 'src' ? v : v)).join(' ')
+        return `${camelToKebab(directive)}=(${allowlist})`
+      }
+      return ''
+    })
+    .filter(Boolean)
+    .join(', ')
+}
+
+function camelToKebab(str: string): string {
```
Comment (watany-dev) [inline src/middleware/secure-headers/secure-headers.ts:294]: By converting uppercase letters that follow lowercase letters or numbers, the implementation becomes simpler
```ts
function camelToKebab(str: string): string {
  return str.replace(/([a-z\d])([A-Z])/g, '$1-$2').toLowerCase()
}
```

File: src/middleware/secure-headers/secure-headers.ts [hunk 2]
```diff
@@ -255,6 +272,36 @@ function getCSPDirectives(
       ]
 }
 
+function getPermissionsPolicyDirectives(policy: PermissionsPolicyOptions): string {
```
Comment (watany-dev) [inline src/middleware/secure-headers/secure-headers.ts:275]: - Reused camelToKebab(directive) by storing it in a kebabDirective variable, reducing redundant function calls.
- Simplified array handling by removing unnecessary checks for specific values like 'self' and 'src'.
- Optimized empty array handling by using a concise conditional, improving code readability and efficiency.

```ts
function getPermissionsPolicyDirectives(policy: PermissionsPolicyOptions): string {
  return Object.entries(policy)
    .map(([directive, value]) => {
      const kebabDirective = camelToKebab(directive)

      if (typeof value === 'boolean') {
        return `${kebabDirective}=${value ? '()' : 'none'}`
      }

      if (Array.isArray(value)) {
        const allowlist = value.length ? value.join(' ') : ''
        return `${kebabDirective}=(${allowlist})`
      }

      return ''
    })
    .filter(Boolean)
    .join(', ')
}

```

**Comment by kbkn3** [comment]
@watany-dev Thank you for reviewing.

@yusukebe Sorry for the delay in response. Fixed.

**Comment by yusukebe** [comment]
@watany-dev Thank you for reviewing!!

@kbkn3 Last one thing. Can you run `bun run format:fix && bun run lint:fix` to format the code?

**Comment by kbkn3** [comment]
@yusukebe 
My apologies.
One line was left blank.
I have formatted it. Please check it!

**Review by yusukebe — APPROVED** [review]
LGTM!

**Comment by yusukebe** [comment]
@kbkn3 

Thank you! Let's go with it. This is a new feature so that it will be included in the next new minor version, `v4.6.0`. I'll later merge this PR to the `next` branch for the new version. Appreciate your first contribution!

**Comment by kbkn3** [comment]
@yusukebe 
Thank you for your time.

I enjoy developing with `hono`. Thanks for your great efforts.

**Pull request merged**